malariagen_data/anoph/to_plink.py

```diff
@@ -0,0 +1,164 @@
+        min_minor_ac,
+        thin_offset,
+        max_missing_an,
+    ):
+        return f"{output_dir}/{region}.{n_snps}.{min_minor_ac}.{thin_offset}.{max_missing_an}"
+
+    def _biallelic_snps_to_plink(
+        self,
+        *,
+        output_dir,
+        region,
+        n_snps,
+        thin_offset,
+        overwrite,
+        sample_sets,
+        sample_query,
+        sample_indices,
+        site_mask,
+        min_minor_ac,
+        max_missing_an,
+        random_seed,
+        inline_array,
+        chunks,
+    ):
+        # Define output file
+        plink_file_path = self._create_plink_outfile(
+            output_dir=output_dir,
+            region=region,
+            n_snps=n_snps,
+            thin_offset=thin_offset,
+            min_minor_ac=min_minor_ac,
+            max_missing_an=max_missing_an,
+        )
+
+        bed_file_path = f"{plink_file_path}.bed"
+
+        # Check to see if file exists and if overwrite is set to false, return existing file
+        if os.path.exists(bed_file_path):
+            if not overwrite:
+                return plink_file_path
+
+        # Get snps
+        ds_snps = self.biallelic_snp_calls(
+            region=region,
+            sample_sets=sample_sets,
+            sample_query=sample_query,
+            sample_indices=sample_indices,
+            site_mask=site_mask,
+            min_minor_ac=min_minor_ac,
+            max_missing_an=max_missing_an,
+            n_snps=n_snps,
+            thin_offset=thin_offset,
+            random_seed=random_seed,
+            inline_array=inline_array,
+            chunks=chunks,
+        )
+
+        # Set up dataset with required vars for plink conversion
+
+        # Compute gt ref counts
+        with self._dask_progress("Computing genotype ref counts"):
+            gt_asc = ds_snps["call_genotype"].data  # dask array
+            gn_ref = allel.GenotypeDaskArray(gt_asc).to_n_ref(fill=-127)
+            gn_ref = gn_ref.compute()
+
+        # Ensure genotypes vary
+        loc_var = np.any(gn_ref != gn_ref[:, 0, np.newaxis], axis=1)
+
+        # Load final data
+        ds_snps_final = dask_compress_dataset(ds_snps, loc_var, dim="variants")
+
+        # Init vars for input to bed reader
+        gn_ref_final = gn_ref[loc_var]
+        val = gn_ref_final.T
+        with self._spinner("Prepare output data"):
+            alleles = ds_snps_final["variant_allele"].values
+            properties = {
+                "iid": ds_snps_final["sample_id"].values,
+                "chromosome": ds_snps_final["variant_contig"].values,
+                "bp_position": ds_snps_final["variant_position"].values,
+                "allele_1": alleles[:, 0],
+                "allele_2": alleles[:, 1],
+            }
+
+        bed_reader.to_bed(
+            filepath=bed_file_path,
+            val=val,
+            properties=properties,
+            count_A1=True,
+        )
+
+        return plink_file_path
+
+    def biallelic_snps_to_plink(
```

Comment:
Please could you add a docstring to this function. We usually use the `@doc` annotation from the `numpydoc_decorator` package to generate the docstrings, if you look at some other modules there should be some examples to follow.

Reply:
Once you have added the docstring, please could you add this function to the API docs at `docs/source/Ag3.rst` and `docs/source/Af1.rst` so that the new function will appear in the [API docs website](https://malariagen.github.io/malariagen-data-python/latest/). It would probably make sense to add the new function under the "SNP data access" section.